_core/page/head.php

```diff
@@ -96,6 +96,73 @@ function generate() {
 
         return $dat;
     }
+    
+    function generateAdmin() {
```

Comment:
A lot of repeated code that could probably be handled by the page class (specifically looking at that extra CSS).